### PR TITLE
Drop beam dimension

### PR DESCRIPTION
**Pull request by emiliom**
I'll leave this PR as a draft. For now, it's meant only for discussion, especially for @leewujung to see the minimal changes I made that fully remove the "unnecessary" use of the `beam` dimension for EK60 and AZFP data. ~~only, for now; I'll add AZFP later, where we're also dropping the `beam` dimension completely.~~ For EK80, the removal will not be complete, as the `beam` dimension will remain for a few variables.

- [x] EK60 code updates
- [x] EK60 test updates
- [x] AZFP code updates
- [x] AZFP test updates
- [x] EK80 code updates
- [x] EK80 test updates

Addresses #978

As this will be a breaking change, our intent is to release it with version 0.8.0.

### CODE REVIEW

File: echopype/convert/set_groups_ek60.py [hunk 1]
```diff
@@ -45,6 +47,9 @@ class SetGroupsEK60(SetGroupsBase):
         "gain_correction",
     }
 
+    # Variables that need beam and ping_time dimensions added to them.
+    beam_ping_time_names = set()
```
Comment (leewujung) [inline echopype/convert/set_groups_ek60.py:51]: This can be removed?

**Review by leewujung — COMMENTED** [review]
@emiliom : I checked the changes and they look fine. I think it makes more sense to review this once all changes for different sonar_models are in, since there are some redundant code in `set_groups_base.py` that needs changing too (I just found out). 

Also, a quick note that please make sure for EK80 CW mode data the `beam` dimension is removed too, since it is identical to the EK60 setup. Thanks!

**Comment by leewujung** [comment]
Related to this, I just did #1057 to capture the other half of what #978 exposed but not specifically called out. The artificial expansion (repetition of the same value) along the `ping_time` dimension is more costly than the length=1 `beam` dimension when data is loaded into memory, and I think we should consider removing it at v0.8.0.

File: echopype/convert/set_groups_ek60.py (hunk 1 above)
Comment (emiliom) [reply echopype/convert/set_groups_ek60.py:51]: For now, it needs to be there. The variable is passed to `set_groups_base.beam_groups_to_convention`, which expects a `set` type.

**Comment by emiliom** [comment]
- Removed the `beam` dimension from AZFP
- Updated all EK60 and AZFP tests to account for the removal of the beam dimension

**Comment by emiliom** [comment]
Note: It's quite possible that in calibration modules there are some checks for the presence of the `beam` dimension that are no longer necessary, and I didn't catch that. For example, this block:

https://github.com/OSOceanAcoustics/echopype/blob/c5afb6521a30abce1d74d0a53edc807f0fae6567/echopype/calibrate/calibrate_ek.py#L579-L584

File: echopype/calibrate/cal_params.py [hunk 2]
```diff
@@ -497,9 +497,14 @@ def _get_fs():
...
-                            beam["frequency_nominal"] / freq_center
-                        )
+                        if "beam" in beam[p].dims:
+                            out_dict[p] = beam[p].isel(beam=0).drop("beam") + 20 * np.log10(
+                                beam["frequency_nominal"] / freq_center
+                            )
+                        else:
+                            out_dict[p] = beam[p] + 20 * np.log10(
+                                beam["frequency_nominal"] / freq_center
+                            )
```
Comment (leewujung) [inline echopype/calibrate/cal_params.py:507]: remove the added if-else as it should not be there

File: echopype/calibrate/calibrate_ek.py [hunk 3]
```diff
@@ -124,9 +124,12 @@ def _cal_power_samples(self, cal_type: str) -> xr.Dataset:
...
 
-        # Squeeze out the beam dim
-        # doing it here because both out and self.cal_params["equivalent_beam_angle"] has beam dim
-        return out.squeeze("beam", drop=True)
+        # Squeeze out the beam dim, if present.
+        # Doing it here because both out and self.cal_params["equivalent_beam_angle"] have` beam dim
+        if "beam" in out.dims:
+            return out.squeeze("beam", drop=True)
+        else:
+            return out
```
Comment (leewujung) [inline echopype/calibrate/calibrate_ek.py:132]: once that the above added if-else is gone, this can be removed.

File: echopype/convert/set_groups_azfp.py [hunk 4]
```diff
@@ -19,13 +19,18 @@ class SetGroupsAZFP(SetGroupsBase):
...
-    beam_only_names = {"backscatter_r"}
+    beam_only_names = set()
 
     # Variables that need only the ping_time dimension added to them.
-    ping_time_only_names = {"sample_interval", "transmit_duration_nominal"}
+    ping_time_only_names = {
+        "sample_interval",
+        "transmit_duration_nominal",
+        "equivalent_beam_angle",
+        "gain_correction",
```
Comment (leewujung) [inline echopype/convert/set_groups_azfp.py:29]: Why are these added? These should be constant throughout `ping_time`.

Comment (leewujung) [reply echopype/convert/set_groups_azfp.py:29]: Ok I see they are added since you remove them from the block below.

File: echopype/tests/echodata/test_echodata.py [hunk 5]
```diff
@@ -459,12 +459,16 @@ def test_nan_range_entries(range_check_files):
             echodata, env_params=None, cal_params=None, ecs_file=None, waveform_mode="BB", encode_mode="complex",
         )
         range_output = cal_obj.range_meter
-        nan_locs_backscatter_r = ~echodata["Sonar/Beam_group1"].backscatter_r.isel(beam=0).drop("beam").isnull()
+        if "beam" in echodata["Sonar/Beam_group1"].dims:
+            nan_locs_backscatter_r = ~echodata["Sonar/Beam_group1"].backscatter_r.isel(beam=0).drop(
+                "beam").isnull()
+        else:
+            nan_locs_backscatter_r = ~echodata["Sonar/Beam_group1"].backscatter_r.isnull()
```
Comment (leewujung) [inline echopype/tests/echodata/test_echodata.py:466]: this is not needed as the file is EK80 broadband complex data -- the `beam` dimension always needs to be dropped.

Comment (leewujung) [reply echopype/tests/echodata/test_echodata.py]: Done

**Comment by leewujung** [comment]
I've caught the cases in calibration code where the `beam` dim already does not exist and hence does not need to be dropped anymore. Also reverted the if-else handling related to `equivalent_beam_angle`, which should not have a `beam` dimension to begin with.

There are few more cases where the `beam` dimension handling needs to be removed:
- `cal_parasm.py::_get_interp_da`
- `test_cal_params.py::beam_AZFP`
- `test_cal_params.py::test_get_interp_da` test function inputs

**Comment by leewujung** [comment]
I've added the remaining changes. Will close this now to move forward to #1057.

**Comment by emiliom** [comment]
Great, thanks!!
